File: packages/zynth-media-library/android/ZynthMediaLibrary/src/main/java/dev/zynth/medialibrary/ZynthMediaLibrary.kt
package dev.zynth.medialibrary

import androidx.activity.ComponentActivity
import com.zynth.kit.runtime.ZynthRuntime

object ZynthMediaLibrary {
    @JvmStatic
    fun initialize(activity: ComponentActivity, runtime: ZynthRuntime) {
        runtime.installModules(listOf(MediaLibraryModule(activity)))
    }
}

File: packages/zynth-media-library/ios/ZynthMediaLibrary.swift
import Foundation
import ZynthKit

@objc(ZynthMediaLibrary)
public class ZynthMediaLibrary: NSObject {
  private static var moduleInstance: MediaLibraryModule?

  @objc public static func initialize(with runtime: ZynthRuntime) {
    guard moduleInstance == nil else {
      print("[ZynthMediaLibrary] Module already initialized")
      return
    }

    let module = MediaLibraryModule(runtime: runtime)
    runtime.installModules([module])
    moduleInstance = module
    print("[ZynthMediaLibrary] Module initialized")
  }

  @objc public static func cleanup() {
    moduleInstance = nil
  }
}
